src/System.Drawing.Primitives/tests/SizeFTests.netcoreapp.cs

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using Xunit;
+
+namespace System.Drawing.PrimitivesTest
+{
+    public partial class SizeFTests
+    {
+        #region SizeF * float tests
+
+        private void MultSizeFFloatTester(float width, float height, float multiplier)
+        {
+            SizeF sz = new SizeF(width, height);
+            SizeF mulExpected;
+
+            mulExpected = new SizeF(width * multiplier, height * multiplier);
+
+            Assert.Equal(mulExpected, sz * multiplier);
+            Assert.Equal(mulExpected, multiplier * sz);
+        }
+
+        [Theory]
+        [InlineData(1000.234f, 2400.933f)]
+        [InlineData(float.MaxValue, 0.0f)]
+        [InlineData(float.MaxValue, 1.0f)]
+        [InlineData(float.MaxValue, -1.0f)]
+        [InlineData(float.MaxValue, float.MaxValue)]
+        [InlineData(float.MinValue, 0.0f)]
+        [InlineData(float.MinValue, 1.0f)]
+        [InlineData(float.MinValue, -1.0f)]
+        [InlineData(float.MinValue, float.MinValue)]
+        [InlineData(float.MaxValue, float.MinValue)]
+        [InlineData(float.MinValue, float.MaxValue)]
+        public void MultiplicationTest(float value1, float value2)
+        {
+            MultSizeFFloatTester(value1, 1.0f, value2);
```

Comment:
The `Dictionary` type is far more complicated than this simple structure, so it only makes sense that it will have far more complicated test cases. I think there's just a few small things that could be cleaned up in the current version:

* `MultiplicationTest` has two vaguely-named parameters (value1 and value2), which are used in a few combinations of ways through a helper class `MultSizeFFloatTester`. If I add `InlineData` to the test case, I feel that it is difficult to immediately assess what the outcome of that will be without looking through all 4 call-sites. Instead, it would be clearer if `MultSizeFFloatTester` was inlined into `MutliplicationTest`, and all variants passed into it through `InlineData`. That involves less code paths to follow, less parameter combinations to grok, and will allow clearer parameter names to be used.
* Again, there are no test cases in which one of the fields is not 1.0f. It is true that, through our prior intuition about how the library code is written, this cannot matter. But from a test perspective it is odd, and hints that we are trying to verify some property that we are not. If you want to ensure that you are not returning `Width * multiplier` for both fields (you mentioned that), then you can just have a test case which passes different values for `width` and `height`.

I hope this doesn't sound needlessly pedantic. As the maintainer of the code, I need to make sure that this code is instantly understandable when I come back to it in several months (or longer), and if I or others need to add test cases to it, it should be as easy as possible for a type like this.